### DISCUSSION TITLE
Can Crossplane be used without clouds?

### DISCUSSION THREAD

**Discussion by Drugoy** [crossplane/crossplane#3202] [Q&A] [answered]
Hello. I've recently stumbled upon Crossplane and I am still not sure I get how it works.
All of our infrastructure is on-premises:
  - k8s cluster (not provisioned by Crossplane)
  - Postgres (outside k8s)
  - MinIO (inside k8s)

Now we are a service company and to provide the service for our customers - we deploy a set of microservices into k8s but they also require a personal DB in Postgres cluster and a personal s3 bucket in MinIO.

We got tired of manually creating Postgres DBs and MinIO buckets and would like to create them at first (and delete them later, as some customers come and go) via some kind of virtual resource in k8s.

Can Crossplane help us do that?
All the docs and intro videos seem to imply that Crossplane requires having a cloud-managed k8s install for some reason - is that so? or can Crossplane be integrated into a pre-existing k8s-cluster running on-premises?

**Comment by cwilhit** [comment, marked as answer]
Yes, it can! But **caveat** later down in my answer.

You can install Crossplane into any Kubernetes cluster. It can be on-prem in a Kubernetes cluster you've created on your own, it can be into a [local Kubernetes cluster](https://kind.sigs.k8s.io/) set up on your own machine, or into one of the many managed Kubernetes services available in the public cloud.

You'd follow the [Getting Started guide](https://crossplane.io/docs/v1.9/getting-started/install-configure.html) like normal. The point where it becomes less relevant for your use case is when the quickstart transitions to explaining how to install one of `providers` into your control plane--specifically referring to [this section](https://crossplane.io/docs/v1.9/getting-started/install-configure.html#install-configuration-package) of the doc. 

So, cutting away at that juncture, what you'd want to look at next is whether there is already a provider authored by the community ([like this list](https://crossplane.io/docs/v1.9/concepts/providers.html)) that implements the crossplane primitives/definitions that you require in order for your newly-minted control plane to be able to instantiate those resources. T**o cut to the short of it, for what you're looking for specifically, I don't think such a provider exists yet**, so you'd want to investigate what it looks like to build this.

**Comment by Drugoy** [comment]
Thanks for the answer!

I wish the docs would provide an extra scenario describing how to install Crossplane to control own on-premises k8s cluster.
For quite many people/companies/cases a managed k8s is not an option at all.
